code/game/objects/items/devices/lightreplacer.dm

```diff
@@ -62,46 +62,51 @@
 	. = ..()
 	. += status_string()
 
+/obj/item/lightreplacer/pre_attack(atom/target, mob/living/user, params)
+	..()
```

Comment:
Respect parent return value?

Reply:
No actually. we want the return value of only `do_action()` proc[which returns TRUE only if the target is a floor, light, floodlight]. I only call the parent proc because maybe it does other stuff behind the scene?

Reply:
Yes, the parent proc sends a signal, and if it returns `TRUE`, you shouldn't do your pre attack. That's what the signal is for. 
You should respect parent return value

Reply:
Done